Rakefile

```diff
@@ -23,7 +33,116 @@ task :travis do
   Rake::Task['spec'].invoke
 end
 
+module HarnessOptions
+  defaults = {
+      :tests  => ['tests'],
+      :log_level => 'debug',
+      :preserve_hosts => 'onfail',
+  }
+
+  DEFAULTS = defaults
+
+  def self.get_options(file_path)
+    puts "Attempting to merge config file: #{file_path}"
+    if File.exists? file_path
+      options = eval(File.read(file_path), binding)
+    else
+      puts "No options file found at #{File.expand_path(file_path)}... skipping"
+    end
+    options || {}
+  end
+
+  def self.get_local_options
+    get_options("./local_options.rb")
+  end
+
+  def self.final_options(intermediary_options = {})
+    local_overrides = get_local_options
+    final_options = DEFAULTS.merge(intermediary_options)
+    final_options.merge!(local_overrides)
+    return final_options
```

Comment:
Nitpick, but trailing `return`s aren't necessary in ruby -- last expression will be the return value from the method.